[Amazon Bedrock Guardrails] Update dashboard with policy violation details

agithomas opened the pull request:
- Enhancement


## Proposed commit message

Update the dashboard with policy violation details including violation code, confidence score, and filter threshold. Update guardrails invocation logs view column titles

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).
- [x] I have verified that any added dashboard complies with Kibana's [Dashboard good practices](https://docs.elastic.dev/ux-guidelines/data-viz/dashboard-good-practices) 

## Author's Checklist

- [x] Dashboard content verification
- [x] Integration testing 

## How to test this PR locally

- `elastic-package build`
- `elastic-package stack up -v -d --services package-registry`

## Related issues

- Closes https://github.com/elastic/integrations/issues/12786 

## Screenshots

**Updated the piechart with details of count of guardrails without violations and guardrails with violations** 

![amazon_bedrock_guardrails_overview](https://github.com/user-attachments/assets/2ad6828c-69be-4d2d-b33b-891bf108e2ed)


**Updated log details view with modified column title, new key information related to guardrails actions and thresholds**

![amazon_bedrock_guardrails_invocation_details](https://github.com/user-attachments/assets/8f4e2e54-c965-4d53-ad85-36847fa7a855)

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/22303)
* Commit: 6182def921f4f2b3add825783525fe793265ef6d

cc @agithomas

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/22303","number":22303,"commit":"6182def921f4f2b3add825783525fe793265ef6d"}],"number":22303}
buildkite-pr-comment-->

shmsr commented on packages/aws_bedrock/kibana/dashboard/aws_bedrock-14fd745a-d3c1-4ebe-bd25-00b465336cde.json:
```diff
@@ -319,17 +414,25 @@
+                                                                    "language": "kuery",
+                                                                    "query": "gen_ai.guardrail_id : * and gen_ai.compliance.violation_detected : true "
+                                                                },
+                                                                "label": "With guardrail interventions (With violations)"
```
```suggestion
                                                                "label": "With guardrail interventions (with violations)"
```

shmsr commented on packages/aws_bedrock/kibana/dashboard/aws_bedrock-14fd745a-d3c1-4ebe-bd25-00b465336cde.json:
```diff
@@ -262,6 +262,100 @@
+                        "type": "lens",
+                        "visualizationType": "lnsMetric"
+                    },
+                    "description": "The latency of the invocations",
```
grammatical mistake:
```suggestion
                    "description": "The latencies of the invocations",
```

agithomas replied:
In the context of the dashboard title, I guess the "The latency of ..." would be the right usage

shmsr commented on packages/aws_bedrock/kibana/dashboard/aws_bedrock-14fd745a-d3c1-4ebe-bd25-00b465336cde.json:
```diff
@@ -262,6 +262,100 @@
+            {
+                "embeddableConfig": {
+                    "attributes": {
+                        "description": "The sum of guardrail policy invocations grouped by guardrail policy name and policy version",
```
```suggestion
                        "description": "The sum of guardrail policy invocations grouped by policy name and version",
```

agithomas replied:
This is an existing description.
Previously i added the word "guardrails" intentionally to avoid confusion between policy types (word policy, PII data policy, etc) from guardrails policy.

shmsr commented:
I did check the dashboards, looks good to me. But I saw couple of cases there were not consistent and then I started checking *.json directly. Here are some following suggestion that we can either take up in this PR or in separate PR so that we maintain sentence case properly (there were many trailing spaces in the README which I have removed through a tool) everywhere:

```diff
diff --git a/packages/aws_bedrock/_dev/build/docs/README.md b/packages/aws_bedrock/_dev/build/docs/README.md
index 78235846ed..8bf26c96cc 100644
--- a/packages/aws_bedrock/_dev/build/docs/README.md
+++ b/packages/aws_bedrock/_dev/build/docs/README.md
@@ -27,7 +27,7 @@ Data streams:
    data for all invocations in your AWS account used in Amazon Bedrock.
  - `runtime`: Collects Amazon Bedrock runtime metrics such as model invocation
    count, invocation latency, input token count, output token count and many
-   more.   
+   more.
  - `guardrails`: Collects Amazon Bedrock Guardrails metrics such as guardrail invocation
 count, guardrail invocation latency, text unit utilization count, guardrail policy types associated with interventions and many more.
 
@@ -62,7 +62,7 @@ either S3 or CloudWatch. For more details, check the
 [Amazon Bedrock User Guide](https://docs.aws.amazon.com/bedrock/latest/userguide/model-invocation-logging.html).
 
 1. Set up an [Amazon S3](https://docs.aws.amazon.com/bedrock/latest/userguide/model-invocation-logging.html#setup-s3-destination) or [CloudWatch Logs](https://docs.aws.amazon.com/bedrock/latest/userguide/model-invocation-logging.html#setup-cloudwatch-logs-destination) destination.
-2. Enable logging. You can do it either through the [Amazon Bedrock console](https://docs.aws.amazon.com/bedrock/latest/userguide/model-invocation-logging.html#model-invocation-logging-console) or [the Amazon Bedrock API](https://docs.aws.amazon.com/bedrock/latest/userguide/model-invocation-logging.html#using-apis-logging). 
+2. Enable logging. You can do it either through the [Amazon Bedrock console](https://docs.aws.amazon.com/bedrock/latest/userguide/model-invocation-logging.html#model-invocation-logging-console) or [the Amazon Bedrock API](https://docs.aws.amazon.com/bedrock/latest/userguide/model-invocation-logging.html#using-apis-logging).
 
 ## Logs
 
@@ -70,9 +70,9 @@ either S3 or CloudWatch. For more details, check the
 
 When log collection from an S3 bucket is enabled, you can access logs from S3 objects referenced by S3 notification events received through an SQS queue or by directly polling the list of S3 objects within the bucket.
 
-The use of SQS notification is preferred: polling list of S3 objects is 
-expensive in terms of performance and costs and should be used only 
-when no SQS notification can be attached to the S3 buckets. This input 
+The use of SQS notification is preferred: polling list of S3 objects is
+expensive in terms of performance and costs and should be used only
+when no SQS notification can be attached to the S3 buckets. This input
 integration also supports S3 notification from SNS to SQS.
 
 To enable the SQS notification method, set the `queue_url` configuration value. To enable the S3 bucket list polling method, configure both the `bucket_arn` and number_of_workers values. Note that `queue_url` and `bucket_arn` cannot be set simultaneously, and at least one of these values must be specified.
@@ -89,10 +89,10 @@ When CloudWatch log collection is enabled, you can retrieve logs from all log st
 
 Amazon Bedrock runtime metrics include `Invocations`, `InvocationLatency`,
 `InvocationClientErrors`, `InvocationServerErrors`, `OutputTokenCount`,
-`OutputImageCount`, `InvocationThrottles`. These metrics can be used for various use cases including: 
- - Comparing model latency 
+`OutputImageCount`, `InvocationThrottles`. These metrics can be used for various use cases including:
+ - Comparing model latency
  - Measuring input and output token counts
- - Detecting the number of invocations that the system throttled   
+ - Detecting the number of invocations that the system throttled
 
 {{event "runtime"}}
 {{fields "runtime"}}
diff --git a/packages/aws_bedrock/kibana/dashboard/aws_bedrock-14fd745a-d3c1-4ebe-bd25-00b465336cde.json b/packages/aws_bedrock/kibana/dashboard/aws_bedrock-14fd745a-d3c1-4ebe-bd25-00b465336cde.json
index 742fdd93c4..2728e3423b 100644
--- a/packages/aws_bedrock/kibana/dashboard/aws_bedrock-14fd745a-d3c1-4ebe-bd25-00b465336cde.json
+++ b/packages/aws_bedrock/kibana/dashboard/aws_bedrock-14fd745a-d3c1-4ebe-bd25-00b465336cde.json
@@ -292,7 +292,7 @@
                                                         "query": "aws_bedrock.guardrails.guardrail_arn : * "
                                                     },
                                                     "isBucketed": false,
-                                                    "label": "Average Guardrails API latency",
+                                                    "label": "Average guardrails API latency",
                                                     "operationType": "average",
                                                     "params": {
                                                         "emptyAsNull": true,
@@ -382,7 +382,7 @@
                                                     "customLabel": true,
                                                     "dataType": "number",
                                                     "isBucketed": false,
-                                                    "label": "Invocations Count",
+                                                    "label": "Invocations count",
                                                     "operationType": "count",
                                                     "params": {
                                                         "emptyAsNull": true,
@@ -937,7 +937,7 @@
                     "y": 18
                 },
                 "panelIndex": "b306f023-4026-450f-a778-1302c59d0286",
-                "title": "Significant Intervened Policy Type (Input \u0026 Output)",
+                "title": "Significant intervened policy type (Input \u0026 Output)",
                 "type": "lens"
             },
             {
@@ -1620,7 +1620,7 @@
                     "y": 60
                 },
                 "panelIndex": "31a89177-77f5-4a30-8976-cbb01e142828",
-                "title": "Guardrails Details",
+                "title": "Guardrails details",
                 "type": "search"
             },
             {
@@ -2250,7 +2250,7 @@
                         "type": "lens",
                         "visualizationType": "lnsXY"
                     },
-                    "description": "Adding guardrails could result in increased latency. The extra latency will depend on the input and response lengths and the guardrails’ implementation and configuration.",
+                    "description": "Adding guardrails could result in increased latency. The extra latency will depend on the input and response lengths, as well as the guardrails' implementation and configuration.",
                     "enhancements": {}
                 },
                 "gridData": {
@@ -3039,4 +3039,4 @@
     "type": "dashboard",
     "typeMigrationVersion": "10.2.0",
     "updated_by": "u_mGBROF_q5bmFCATbLXAcCwKa0k8JvONAwSruelyKA5E_0"
-}
\ No newline at end of file
+}
diff --git a/packages/aws_bedrock/kibana/dashboard/aws_bedrock-2a19b571-251b-487b-84b2-abd887efb8a4.json b/packages/aws_bedrock/kibana/dashboard/aws_bedrock-2a19b571-251b-487b-84b2-abd887efb8a4.json
index 359be1e613..2266d906c3 100644
--- a/packages/aws_bedrock/kibana/dashboard/aws_bedrock-2a19b571-251b-487b-84b2-abd887efb8a4.json
+++ b/packages/aws_bedrock/kibana/dashboard/aws_bedrock-2a19b571-251b-487b-84b2-abd887efb8a4.json
@@ -145,7 +145,7 @@
                     "y": 0
                 },
                 "panelIndex": "a032aba8-cb49-443c-a985-5973689ba62a",
-                "title": "Related Dashboards",
+                "title": "Related dashboards",
                 "type": "links"
             },
             {
@@ -1161,7 +1161,7 @@
                                                         "query": "aws_bedrock.runtime.model_id : * "
                                                     },
                                                     "isBucketed": false,
-                                                    "label": "Invocation Latency",
+                                                    "label": "Invocation latency",
                                                     "operationType": "sum",
                                                     "params": {
                                                         "emptyAsNull": true
@@ -1898,7 +1898,7 @@
                                                         "query": "aws_bedrock.runtime.model_id : *  and NOT aws_bedrock.runtime.quality:* and not aws_bedrock.runtime.image_size : *"
                                                     },
                                                     "isBucketed": false,
-                                                    "label": "Output Images",
+                                                    "label": "Output images",
                                                     "operationType": "sum",
                                                     "params": {
                                                         "emptyAsNull": true,
@@ -2043,7 +2043,7 @@
                                                         "query": "aws_bedrock.runtime.model_id : *  and NOT aws_bedrock.runtime.quality:* and not aws_bedrock.runtime.image_size : *"
                                                     },
                                                     "isBucketed": false,
-                                                    "label": "Output Images",
+                                                    "label": "Output images",
                                                     "operationType": "sum",
                                                     "params": {
                                                         "emptyAsNull": true,
@@ -2533,7 +2533,7 @@
                                     "mode": "full",
                                     "niceValues": true
                                 },
-                                "yTitle": "Response Time"
+                                "yTitle": "Response time"
                             }
                         },
                         "title": "",
@@ -2726,7 +2726,7 @@
                                     "mode": "full",
                                     "niceValues": true
                                 },
-                                "yTitle": "Percentage of Time Taken"
+                                "yTitle": "Percentage of time taken"
                             }
                         },
                         "title": "",
@@ -2859,4 +2859,4 @@
     ],
     "type": "dashboard",
     "typeMigrationVersion": "10.2.0"
-}
\ No newline at end of file
+}
```

shmsr commented:
Rest looks good!

agithomas commented:
>  Also see the screenshot, the "G" of "Average Guardrails API latency" is not consistent with the just the previous (lhs) panel:
Many of these changes are based on the feedback received.

agithomas commented:
@shmsr,

I have addressed most of the feedback. For the comments mentioned [here](https://github.com/elastic/integrations/pull/12787#issuecomment-2658687139), this we could take it up in future along with other planned changes.

Could you please take one final look at the PR and approve the changes if all looks good?

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/22322)
* Commit: 4844a2412630106f40176b20f467a2ff893e69c9

### History
* :green_heart: [Build #22303](https://buildkite.com/elastic/integrations/builds/22303) succeeded 6182def921f4f2b3add825783525fe793265ef6d

cc @agithomas

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/22322","number":22322,"commit":"4844a2412630106f40176b20f467a2ff893e69c9"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/22303","number":22303,"commit":"6182def921f4f2b3add825783525fe793265ef6d"}],"number":22322}
buildkite-pr-comment-->
